### ISSUE TITLE
Add support for .NEF raw photos

### ISSUE THREAD

**Issue by photopea** [photopea/photopea#955]

**Comment by Greenplugin** [comment]
It will be great!

**Comment by photopea** [comment]
@Greenplugin I can not find any specification of the NEF compression anywhere on the internet.

I only found implementations: https://www.cybercom.net/~dcoffin/dcraw/dcraw.c - nikon_load_raw, but it is too hard to understand the format from their code. I can not find anybody to help me :(

**Comment by TomasHubelbauer** [comment]
Did you try contacting the author of that document on their email for help? They may be willing to give you a few pointers since you're showing interest in their work.

**Comment by photopea** [comment]
Yes, I contacted two authors of two different projects, without any response :(

**Comment by Greenplugin** [comment]
> Yes, I contacted two authors of two different projects, without any response :(

what project authors have you contacted?

**Comment by photopea** [comment]
@Greenplugin  Dave Coffin, he has his email at the top of dcraw.c - dcoffin@cybercom.net

Also, Hubert Figuière @hfiguiere, the author of libopenraw: https://libopenraw.freedesktop.org/wiki/Nikon_NEF/

**Comment by hfiguiere** [comment]
Why don't you just use one of the libraries that exists instead of trying to reinvent the wheel?

**Comment by hfiguiere** [comment]
(there is also libraw and rawspeed)

**Comment by photopea** [comment]
@hfiguiere We need the code to be Javascript, so it can run in a browser. I am not familiar with C to Javascript converters, but I think reimplementing it would be simpler, than studying these converters.

We already have full DNG support at www.Photopea.com .

Also, I think that a public description of NEF format would be a benefit for the whole humanity. Just like we publicly described the CDR format: https://github.com/photopea/CDR-specification

**Comment by photopea** [comment]
Maybe @yetithefoot could help us write the public description of .NEF compression format? I already know how to build a Huffman tree for 12 and 14-bit depths, but I am not sure how the Predictor works.
